adaptive/learner/averagelearner1D.py

```diff
@@ -0,0 +1,399 @@
+            self._update_rescaled_error_in_mean(x, "resampled")
+
+            if self._error_in_mean[x] <= self.min_Delta_g or n >= self.max_samples:
+                self._rescaled_error_in_mean.pop(x, None)
+
+            # We also need to update scale and losses
+            super()._update_scale(x, y)
+            self._update_losses_resampling(x, real=True)  #  XXX: REVIEW
+
+            # XXX: Is the following necessary?
+            # If the scale has increased enough, recompute all losses.
+            if self._scale[1] > self._recompute_losses_factor * self._oldscale[1]:
+                for interval in reversed(self.losses):
+                    self._update_interpolated_loss_in_interval(*interval)
+                self._oldscale = deepcopy(self._scale)
+
+    def _update_distances(self, x):
+        x_left, x_right = self.neighbors[x]
+        y = self.data[x]
+        if x_left is not None:
+            self._distances[x_left] = hypot((x - x_left), (y - self.data[x_left]))
+        if x_right is not None:
+            self._distances[x] = hypot((x_right - x), (self.data[x_right] - y))
+
+    def _update_losses_resampling(self, x, real=True):
+        """Update all losses that depend on x, whenever the new point is a
+           re-sampled point"""
+        # (x_left, x_right) are the "real" neighbors of 'x'.
+        x_left, x_right = self._find_neighbors(x, self.neighbors)
+        # (a, b) are the neighbors of the combined interpolated
+        # and "real" intervals.
+        a, b = self._find_neighbors(x, self.neighbors_combined)
+
+        if real:
+            for ival in _get_intervals(x, self.neighbors, self.nth_neighbors):
+                self._update_interpolated_loss_in_interval(*ival)
+        elif x_left is not None and x_right is not None:
+            # 'x' happens to be in between two real points,
+            # so we can interpolate the losses.
+            dx = x_right - x_left
+            loss = self.losses[x_left, x_right]
+            self.losses_combined[a, x] = (x - a) * loss / dx
+            self.losses_combined[x, b] = (b - x) * loss / dx
+
+        # (no real point left of x) or (no real point right of a)
+        left_loss_is_unknown = (x_left is None) or (not real and x_right is None)
+        if (a is not None) and left_loss_is_unknown:
+            self.losses_combined[a, x] = float("inf")
+
+        # (no real point right of x) or (no real point left of b)
+        right_loss_is_unknown = (x_right is None) or (not real and x_left is None)
+        if (b is not None) and right_loss_is_unknown:
+            self.losses_combined[x, b] = float("inf")
+
+    def _calc_error_in_mean(self, ys, y_avg, n):
+        variance_in_mean = sum((y - y_avg) ** 2 for y in ys) / (n - 1)
+        t_student = scipy.stats.t.ppf(1 - self.alpha, df=n - 1)
+        return t_student * (variance_in_mean / n) ** 0.5
+
+    def tell_many(self, xs, ys, *, force=False):
+        """The data should be given as:
```

Comment:
doc-string is incorrect

Reply:
Fixed in [this commit](https://github.com/python-adaptive/adaptive/pull/283/commits/4a1fa919b19ef378eae25b8620668b78f72d66a7)!